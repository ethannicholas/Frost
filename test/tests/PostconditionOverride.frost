@extendable
class PostconditionOverride1 {
    var x:Int

    @extendable
    @post(self.x >= 0)
    method test(x:Int) {
        self.x := x
        Console.printLine(x)
    }
}

class PostconditionOverride2 : PostconditionOverride1 {
    @override
    @postAnd(self.x <= 10)
    method test(x:Int) {
        self.x := x
        Console.printLine(x)
    }

    @class
    method main() {
        PostconditionOverride2().test(1)
        PostconditionOverride2().test(5)
        PostconditionOverride2().test(11)
    }
}