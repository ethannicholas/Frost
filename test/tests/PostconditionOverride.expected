1
5
11
assertion failure (PostconditionOverride.panda:14): postcondition failure for method anonymous.PostconditionOverride2.test(x:panda.core.Int64)
